src/core/ipns/resolver.js

```diff
@@ -111,8 +125,15 @@ class IpnsResolver {
         return callback(errcode(new Error(errMsg), 'ERR_INVALID_RECORD_RECEIVED'))
       }
```

Comment:
Do we need this check here? Does `Record.deserialize`/`ipns.unmarshal` deal with this? - it should do!